src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -551,11 +550,17 @@ class MergeTreeData : public IStorage, public WithMutableContext
     MutableDataPartsVector tryLoadPartsToAttach(const ASTPtr & partition, bool attach_part,
                                                 ContextPtr context, PartsTemporaryRename & renamed_parts);
 
-
     /// If the table contains too many active parts, sleep for a while to give them time to merge.
     /// If until is non-null, wake up from the sleep earlier if the event happened.
     void delayInsertOrThrowIfNeeded(Poco::Event * until, const ContextPtr & query_context) const;
 
+    /// If the table contains too many unfinished mutations, sleep for a while to give them time to execute.
```

Comment:
Only true if `number_of_mutations_to_delay` is set right ?

Reply:
Yes.

Reply:
would be helpful to add this to the comment